test/library/packages/LinearAlgebra/performance/dot-perf.chpl

```diff
@@ -20,35 +19,36 @@ config type eltType = real;
 const nbytes = numBytes(eltType);
 
 proc main() {
-  const Space = {1..n};
-  const BlockSpace = Space dmapped Block(boundingBox=Space);
+  const dom = {1..n, 1..n};
   
-  var BA : [BlockSpace] real = [i in BlockSpace] (i / (1000000));
+  var BA : [dom] real = [(i,j) in dom] (j / (1000000));
 
   var t: Timer;
 
   if !correctness {
     writeln('============================');
     writeln('Dense Dot Performance Test');
     writeln('============================');
-    writeln('iters : ', iters);
-    writeln('n     : ', n);
-    writeln('numLocales     : ', numLocales);
-    writeln('MB    : ', (nbytes*n) / 10**6);
+    writeln('iters                  : ', iters);
+    writeln('square matrix size     : ', n);
+    writeln('MB                     : ', (nbytes*n*n) / 10**6);
     writeln();
   }
-  
-  var d: eltType;
 
+  var result = dot(BA, BA);
+
+  t.start();
   for 1..iters {
-    t.start();
-    d = dot(BA, BA);
-    t.stop();
+    dot(BA, BA);
```

Comment:
Not for this PR, but we should think of ways to make sure that this call is not dead-code-eliminated. If we had an overload that took an output array, that should be used for example.

Reply:
Is there a pragma that can be used to ensure this?

Reply:
Not that I am aware of, nor I would advocate for a pragma in a performance test. As long as we are not eliminating that right now, I think we are fine. We'll notice in the performance plots if it gets eliminated after a compiler change or something. I am not that worried.

Reply:
How about we continuously update `result`? The assignment to `result` could be considered part of the cost of doing the dot product.